### DISCUSSION TITLE
Defining more than two objective in NSGA II with pymoo.

### DISCUSSION THREAD

**Discussion by jitniknita2016** [anyoptimization/pymoo#227] [Q&A]
I am new to pymoo. Working with a problem having more than two objective functions (developed by me). How can I define all the objective functions (more than two) in pymoo for NSGA II.  

The code I got is for 2 objective-  https://pymoo.org/getting_started/part_2.html

class MyProblem(ElementwiseProblem):

    def __init__(self):
        super().__init__(n_var=2,
                         n_obj=2,
                         n_constr=2,
                         xl=np.array([-2,-2]),
                         xu=np.array([2,2]))

    def _evaluate(self, x, out, *args, **kwargs):
        f1 = 100 * (x[0]**2 + x[1]**2)
        f2 = (x[0]-1)**2 + x[1]**2

        g1 = 2*(x[0]-0.1) * (x[0]-0.9) / 0.18
        g2 = - 20*(x[0]-0.4) * (x[0]-0.6) / 4.8

        out["F"] = [f1, f2]
        out["G"] = [g1, g2]

**Comment by blankjul** [comment]
Simply append more values to the objective list. 
Make sure to use an appropriate algorithm for many-objective optimization if you do so.


```
def __init__(self):
    super().__init__(n_var=2,
                     n_obj=2,
                     n_constr=2,
                     xl=np.array([-2,-2]),
                     xu=np.array([2,2]))

def _evaluate(self, x, out, *args, **kwargs):
    f1 = 100 * (x[0]**2 + x[1]**2)
    f2 = (x[0]-1)**2 + x[1]**2
    f3 = ...
    f4 = ...

    g1 = 2*(x[0]-0.1) * (x[0]-0.9) / 0.18
    g2 = - 20*(x[0]-0.4) * (x[0]-0.6) / 4.8

    out["F"] = [f1, f2, f3, f4]
    out["G"] = [g1, g2]

```
